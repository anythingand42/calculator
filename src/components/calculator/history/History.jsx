import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { operations } from '../shared/const';

const StyledTable = styled.table`
  width: 100%;
  border-collapse: collapse;
  th, td {
    border: 1px solid #ddd;
    padding: 4px;
    width: 25%
  }

  td {
    text-align: center;
  }

  tr:nth-child(even){
    background-color: #f2f2f2;
  }

  th {
    background-color: #f2f2f2;
  }
`;

function History({ entries, className }) {
  if (entries.length === 0) return null;
  return (
    <StyledTable className={className}>
      <thead>
        <tr>
          <th>A</th>
          <th>Operation</th>
          <th>B</th>
          <th>Result</th>
        </tr>
      </thead>
      <tbody>
        {entries.map((entry) => (
          <tr>
            <td>{entry.A}</td>
            <td>{operations[entry.operation]}</td>
            <td>{entry.B}</td>
            <td>{entry.result}</td>
          </tr>
        ))}
      </tbody>
    </StyledTable>
  );
}

History.propTypes = {
  entries: PropTypes.arrayOf(PropTypes.shape({
    A: PropTypes.string,
    operation: PropTypes.string,
    B: PropTypes.string,
    result: PropTypes.string,
  })).isRequired,
  className: PropTypes.string,
};

History.defaultProps = {
  className: '',
};

export default History;
