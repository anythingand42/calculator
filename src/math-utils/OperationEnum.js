const OperationEnum = {
  sum: 0,
  divide: 1,
  mod: 2,
  maxPrimeBetween: 3,
};

export default OperationEnum;
