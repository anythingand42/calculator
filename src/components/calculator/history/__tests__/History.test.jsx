import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import History from '../History';
import OperationEnum from '../../../../math-utils/OperationEnum';

test('History component', () => {
  const { container } = render(
    <History
      entries={[
        {
          key: 0, A: '2', B: '2', operation: OperationEnum.sum, result: '4',
        },
        {
          key: 1, A: '2', B: '4', operation: OperationEnum.divide, result: '0.5',
        },
        {
          key: 2, A: '3', B: '2', operation: OperationEnum.mod, result: '1',
        },
        {
          key: 3, A: '2', B: '7', operation: OperationEnum.maxPrimeBetween, result: '7',
        },
      ]}
    />,
  );
  expect(container.firstChild).toMatchSnapshot();
});
