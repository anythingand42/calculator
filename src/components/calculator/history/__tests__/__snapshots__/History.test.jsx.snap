// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component 1`] = `
.c0 {
  width: 100%;
  border-collapse: collapse;
}

.c0 th,
.c0 td {
  border: 1px solid #ddd;
  padding: 4px;
  width: 25%;
}

.c0 td {
  text-align: center;
}

.c0 tr:nth-child(even) {
  background-color: #f2f2f2;
}

.c0 th {
  background-color: #f2f2f2;
}

<table
  class="c0"
>
  <thead>
    <tr>
      <th>
        A
      </th>
      <th>
        Operation
      </th>
      <th>
        B
      </th>
      <th>
        Result
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        2
      </td>
      <td>
        + (sum)
      </td>
      <td>
        2
      </td>
      <td>
        4
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        / (divide)
      </td>
      <td>
        4
      </td>
      <td>
        0.5
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        % (mod)
      </td>
      <td>
        2
      </td>
      <td>
        1
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        max prime between
      </td>
      <td>
        7
      </td>
      <td>
        7
      </td>
    </tr>
  </tbody>
</table>
`;
